systemd/systemd.go

```diff
@@ -103,6 +104,58 @@ type Systemd interface {
 // A Log is a single entry in the systemd journal
 type Log map[string]interface{}
 
+// RestartCondition encapsulates the different systemd 'restart' options
+type RestartCondition string
+
+// These are the supported restart conditions
+const (
+	RestartNever      RestartCondition = "never"
+	RestartOnSuccess  RestartCondition = "on-success"
+	RestartOnFailure  RestartCondition = "on-failure"
+	RestartOnAbnormal RestartCondition = "on-abnormal"
+	RestartOnAbort    RestartCondition = "on-abort"
+	RestartAlways     RestartCondition = "always"
+)
+
+var restartMap = map[string]RestartCondition{
+	"never":       RestartNever,
+	"on-success":  RestartOnSuccess,
+	"on-failure":  RestartOnFailure,
+	"on-abnormal": RestartOnAbnormal,
+	"on-abort":    RestartOnAbort,
+	"always":      RestartAlways,
+}
+
+// ErrUnknownRestartCondition is returned when trying to unmarshal an unknown restart condition
+var ErrUnknownRestartCondition = errors.New("invalid restart condition")
+
+func (rc RestartCondition) String() string {
+	s := string(rc)
+	if s == "" {
+		s = string(RestartOnFailure)
```

Comment:
This feels a bit like a lie.. the rc is not actually a RestartOnFailure, so its string representation shouldn't claim to be it. Imagine us trying to debug an issue where we look at the rc and it just magically changes somewhere.

The actual truth is that we want GenServiceFile to present a default if the rc is unset, so I'd suggest doing that there.